### PR TITLE
Add RECV-OCTETS, SEND now accepts ub8 vectors

### PR DESCRIPTION
**Pull request by phoe**
As in title.

Docs were regenerated using the code from `gendocs.lisp` - I hope I didn't screw anything up.

### CODE REVIEW

File: c-api.lisp [hunk 1]
```diff
@@ -565,6 +564,10 @@ Connected socket may not receive messages sent before it was bound.
            (locally (declare (type (integer 1 #.most-positive-fixnum)
                                    len))
              (%send socket buf (1- len) flags)))))
+      ((typep buf '(vector (unsigned-byte 8)))
+       (with-foreign-array (pointer buf `(:array :unsigned-char ,(length buf)))
```
Comment (phoe) [inline c-api.lisp]: TODO: change to `WITH-POINTER-TO-VECTOR-DATA` to avoid one layer of consing.

File: c-api.lisp [hunk 2]
```diff
@@ -546,10 +546,9 @@ Connected socket may not receive messages sent before it was bound.
 
 (defun send (socket buf &key len dontwait sndmore
                              (encoding cffi:*default-foreign-encoding*))
-  ;; (declare (optimize speed))
```
Comment (orivej) [inline c-api.lisp]: This is not just a forgotten piece of code. I am sure that the intention was for any change in `send` to uncomment this, make sure that the changed code can not be obviously optimized further, and comment it back to hide warnings from the end users.

File: lisp-api.lisp [hunk 3]
```diff
@@ -21,6 +21,15 @@
      (foreign-string-to-lisp (msg-data msg) :count (msg-size msg) :encoding encoding)
      (getsockopt socket :rcvmore))))
 
+(defun recv-octets (socket &key dontwait)
+  "Receive a message part from a socket as an octet vector."
```
Comment (orivej) [inline lisp-api.lisp]: Please describe the return value in the docstring (here and in `recv-string`). (There is a special `@return` syntax for return values.)

File: c-api.lisp (hunk 2 above)
Comment (phoe) [reply c-api.lisp]: OK - I'll rewrite this comment to include this remark, since it was confusing for me.

File: lisp-api.lisp (hunk 3 above)
Comment (phoe) [reply lisp-api.lisp]: OK, will do.

**Comment by phoe** [comment]
In case you'd like to merge both this and #31, can I fix the review comments in a new commit atop these two? Or should I force-push to both branches instead?

**Comment by orivej** [comment]
Let's finish with this PR first, then you could update the other one on top of the new `master`.

**Comment by phoe** [comment]
OK, will do exactly that then.

**Comment by phoe** [comment]
OK, both comments fixed, docs have been regenerated.

The additional commits can be squashed&merged.

File: lisp-api.lisp [hunk 4]
```diff
@@ -14,13 +14,24 @@
            ,@body))))
 
 (defun recv-string (socket &key dontwait (encoding cffi:*default-foreign-encoding*))
-  "Receive a message part from a socket as a string."
+  "Receive a message part from a socket as a string.
+@return{A string.}"
```
Comment (orivej) [inline lisp-api.lisp]: This should be either a plain text description of the two return values, or two `@return` lines — please choose what looks better in HTML.

Comment (phoe) [reply lisp-api.lisp]: Oh right, I forgot about the secondary return value - let me fix that.

**Pull request merged**